File: java1_KDG/Ch05/src/sub03/Car.java
package sub03;

// 클래스 정의
public class Car {
	
	// 자동차의 속성(변수)
	private String name;
	private String color;
	private int speed;
	public static int count;
	
	//생성자
	public Car(String name, String color, int speed) {
		this.name = name;
		this.color = color;
		this.speed =speed;
		count += 1;
	}
	
	// Getter, Setter
	public String getColor() {
		return color;
	}
	public void setColor(String color) {
		this.color = color;
	}
	
	public String getName() {
		return name;
	}
	public void setName(String name) {
		this.name = name;
	}
	
	public int getSpeed() {
		return speed;
	}
	public void setSpeed(int speed) {
		this.speed = speed;
	}
	
	
	// 자동차의 기능(메서드)
	public void speedUp(int speed) {
		this.speed += speed;
	}
	
	public void speedDown(int speed) {
		this.speed -= speed;
	}
	
	public void show() {
		System.out.println("차량명 : "+this.name); // 가독성을 위해 붙여도 됨
		System.out.println("차량색 : "+color);
		System.out.println("속도 : "+speed);
		System.out.println("차량번호 : "+count);
	}
	
}

File: Python/Ch06/sub1/Car.py
class Car:

    #생성자
    def __init__(self, brand, color, price):
        #속성
        self.brand = brand
        self.color = color
        self.price = price
        self.speed = 0
    #기능
    def speedUp(self):
        print('%s 속도 올리기...' % self.brand)
        self.speed += 20
    
    def speedDown(self):
        print('%s 속도 내리기...' % self.brand)
        self.speed -= 20
    
    def show(self):
        print('차량명 :', self.brand)
        print('도색 :', self.color)
        print('가격 :', self.price)
        print('속도 :', self.speed)

